### DISCUSSION TITLE
Paste into selected folder without entering/opening

### DISCUSSION THREAD

**Discussion by Kayzels** [sxyazi/yazi#957] [1. Q&A] [answered]
I don't know if this is possible, or advised. Suppose I have many files, and I want to move them into subfolders of the current directory based on some criteria. So I select the files to be moved.

Then I navigate to where the folder name is in the list, and open it. And then paste.

What I'd like to be able to do would be to navigate to the folder in the list, and paste without opening it. So it should paste in the current folder if a folder isn't selected, and paste in the subfolder if a subfolder is selected.

Or even maybe have a keymap that pastes it in the subfolder? Would this be possible to set up?

At the moment, I keep forgetting to open the subfolder, and then it makes copies rather than moving it.

**Comment by sxyazi** [comment, marked as answer]
It's possible, you can write a plugin in a few lines of code to achieve it:

```lua
return {
  entry = function()
    local h = cx.active.current.hovered
    if h and h.cha.is_dir then
      ya.manager_emit("enter", {})
      ya.manager_emit("paste", {})
      ya.manager_emit("leave", {})
    else
      ya.manager_emit("paste", {})
    end
  end,
}
```

Save it as `~/.config/yazi/plugins/smart-paste.yazi/init.lua`, then bind it to the `p` key in your `keymap.toml`: 

```bash
{ on = [ "p" ], run = "plugin --sync smart-paste" }
```


https://github.com/sxyazi/yazi/assets/17523360/080212b5-43e7-4c36-83e8-312495d50383

**Comment by bertradio** [comment]
And if you want to force overwrite this works:

return {
	entry = function()
		local h = cx.active.current.hovered
		if h and h.cha.is_dir then
			ya.manager_emit("enter", {})
			ya.manager_emit("paste", { force = true })
			ya.manager_emit("leave", {})
		else
			ya.manager_emit("paste", { force = true })
		end
	end,
}
